library/ccm.c

```diff
@@ -403,7 +406,13 @@ static const unsigned char res[NB_TESTS][32] = {
 int mbedtls_ccm_self_test( int verbose )
 {
     mbedtls_ccm_context ctx;
-    unsigned char out[32];
+    /*
+     * Some HW accelerators require the input and output buffers to be aligned
+     * and / or to be contiguous. Define local buffers on the stack, to have contiguous
+     * aligned buffers for the accelerators.
```

Comment:
Now that we understand better the actual problem, that some accelerators require the test vectors to be in RAM (because they can't access flash), we should update this comment. It shouldn't say anything about alignment or continuity, since those weren't the problems we have actually ever seen in practice.